index.js

```diff
@@ -1,41 +1,42 @@
 'use strict';
-var os = require('os');
-
-var type = {
-	v4: {
-		def: '127.0.0.1',
-		family: 'IPv4'
-	},
-	v6: {
-		def: '::1',
-		family: 'IPv6'
-	}
-};
-
-function internalIp(version) {
-	var options = type[version];
-	var ret = options.def;
-	var interfaces = os.networkInterfaces();
-
-	Object.keys(interfaces).forEach(function (el) {
-		interfaces[el].forEach(function (el2) {
-			if (!el2.internal && el2.family === options.family) {
-				ret = el2.address;
-			}
-		});
-	});
 
-	return ret;
-}
+// TODO: Remove dependency on ip module
+// https://github.com/whitequark/ipaddr.js/issues/59
 
-function v4() {
-	return internalIp('v4');
-}
+const os = require('os');
+const defaultGateway = require('default-gateway');
+const ip = require('ip');
+const ipaddr = require('ipaddr.js');
+
+function internalIp(family) {
+	return defaultGateway[family]().then(result => {
+		const interfaces = os.networkInterfaces();
+		let ret;
 
-function v6() {
-	return internalIp('v6');
+		// Remove IPv6 zone index and parse gateway address as a ipaddr.js object
+		// https://github.com/whitequark/ipaddr.js/issues/60
+		const gatewayIp = ipaddr.parse(result.gateway.replace(/%.+/, ''));
+
+		// Look for the matching interface in all local interfaces
+		Object.keys(interfaces).some(name => {
+			return interfaces[name].some(addr => {
+				const subnet = ip.subnet(addr.address, addr.netmask);
+				const net = ipaddr.parseCIDR(addr.address + '/' + subnet.subnetMaskLength);
+				if (net[0].kind() === gatewayIp.kind() && gatewayIp.match(net)) {
+					ret = net[0].toString();
+				}
+				return Boolean(ret);
+			});
+		});
+
+		if (!ret) {
+			throw new Error('Unable to determine internal IP');
+		}
```

Comment:
I just did that, seems better to stay compatible here than to introduce unhelpful errors.